src/app/dts/dts.component.html

```diff
@@ -1,19 +1,73 @@
-<mat-tab-group (selectedTabChange)="tabClick($event)">
-  <mat-tab label="Template">
-    <ngx-datatable
-      class="material striped"
-      [rows]="rows | async"
-      [columns]="columns"
-      [headerHeight]="50"
-      [footerHeight]="50"
-      [rowHeight]="50"
-      (activate)="rowClick($event)"
-      [messages]="{ emptyMessage: 'No templates found.' }"
-      [sorts]="[ {prop: 'author', dir: 'asc'} ]">
-    </ngx-datatable>
-  </mat-tab>
-  <mat-tab label="Edit" [disabled]="selectedTemplate == null">
-    <app-template-editor *ngIf="showEditor" [templateObject]="selectedTemplate">
-    </app-template-editor>
-  </mat-tab>
-</mat-tab-group>
+<form [formGroup]="dtsForm">
+  <button type="button" (click)="newTemplate()" mat-button>
+    <b>&#43;</b>
+  </button>
+  <mat-form-field class="filterby-form-field">
+    <input class="filterBy" matInput type="text" formControlName="templateFilter" placeholder="Type to filter by key or author...">
+    <button mat-button *ngIf="dtsForm.value.templateFilter" matSuffix mat-icon-button aria-label="Clear" (click)="dtsForm.controls['templateFilter'].setValue('')">
```

Comment:
Wouldn't this do the job?
`(click)="dtsForm.reset()"`
Seems safer than to have the string `templateFilter` in there.

Reply:
Implemented change.